feat: allow making multiplication arithmetics with Interval data type

flaneur2020 opened the pull request:
## Changes Made

<!-- Describe what changes were made and why. Include implementation details if necessary. -->

this pr allows executing multiplication arithmetics for interval data type. like:

```py
import daft

daft.sql("select interval '1 day' * 3;").collect()
```

first time submitting a PR here, appreciate the feedback!

## Related Issues

<!-- Link to related GitHub issues, e.g., "Closes #123" -->

fixes #4144

## Checklist

- [ ] Documented in API Docs (if applicable)
- [ ] Documented in User Guide (if applicable)
- [ ] If adding a new documentation page, doc is added to `docs/mkdocs.yml` navigation
- [ ] Documentation builds and is formatted properly (tag @/ccmao1130 for docs review)

universalmind303 reviewed (commented):
Thanks @flaneur2020. 

Overall, this looks pretty good. Can we add some more test cases for this though. I'd suggest adding a few testing each interval type (month,day, year), and test it against all of the numeric types. This should be pretty easy to set up with parametrized pytests. It also looks like you added support for numeric * duration. Lets test that as well!

flaneur2020 commented:
@universalmind303 updated this PR, PTAL 🫡 

- added more tests covering different interval types (years, months, hours).
- removed the changes related to Duration from this PR; they can be addressed in a separate pr.

(there's still one line of coverage left in https://github.com/Eventual-Inc/Daft/pull/4261/files#diff-65364050580a248fc4dbdd6a23d60e68b6f0b39eafa232205a51f5b22a5e6b0dR292, but I couldn't find a reasonable way to cover it with tests as it's already checked in `InferDataType#mul`)

ccmao1130 commented:
Thank you for your contribution @flaneur2020 🎉
